subxt/src/client/online_client.rs

```diff
@@ -90,22 +123,69 @@ impl<T: Config> OnlineClient<T> {
     pub async fn from_rpc_client<R: RpcClientT>(
         rpc_client: Arc<R>,
     ) -> Result<OnlineClient<T>, Error> {
-        let rpc = Rpc::new(rpc_client);
+        OnlineClient::from_rpc_client_at(rpc_client, None).await
+    }
 
+    /// Construct a new [`OnlineClient`] by providing an underlying [`RpcClientT`]
+    /// implementation to drive the connection, as well as a block hash from which to
+    /// obtain the metadata information from.
+    ///
+    /// # Warning
+    ///
+    /// - If you use this function, subxt will be unable to work with blocks that aren't
+    ///   compatible with the metadata at the provided block hash, and if the block hash is
+    ///   not recent, things like subscribing to the head of the chain and submitting
+    ///   transactions will likely fail or encounter errors.
+    /// - Subxt does not support blocks using pre-V14 metadata and will error if you attempt
+    ///   to instantiate a client at such a block.
+    pub async fn from_rpc_client_at<R: RpcClientT>(
+        rpc_client: Arc<R>,
+        block_hash: Option<T::Hash>,
```

Comment:
it's almost like we should remove this `Option stuff` with a enum or something which it's much more obvious than to have `None` which means the latest state/block

Reply:
or perhaps document that `None` implies the latest

Reply:
I don't care too much about Substrate "compatibility" here, so maybe an enum is the way to go, especially with all of the other `at()` methods we have around (my thinking there was to provide `.at()` and `.at_current()` in those places, but maybe it's easier to just do `.at(Block::Current)`/`.at(Block::Hash)` (cc @lexnv).

Anyway, I think it's all good for now but def food for thought :)